### ISSUE TITLE
Error running in SketchUp Pro 2023?

### ISSUE THREAD

**Issue by rmfuhrer** [marvingreenberg/FaceSVG#45]
I'm working in SketchUp Pro 2023 on MacOS Ventura, with the latest FaceSVG plugin (as of today, 18 Oct 2023, that seems to be 3.0.2).

I have a relatively simple design - a 24x32" slab, 1.5" thick, with an 8x6 array of 20mm holes, spaced 96mm apart in both dimensions. I want to cut the holes using Shaper Origin.

I selected the top face, right-clicked and ran "FaceSVG->Layout SVG Profile' and got the following errors in a popup:

reference to deleted Entity
 /Users/rmfuhrer/Library/Application Support/SketchUp 2023/SketchUp/Plugins/facesvg/layout.rb:125:in `transform_entities'
*/Users/rmfuhrer/Library/Application Support/SketchUp 2023/SketchUp/Plugins/facesvg/layout.rb:125:in `block in layout_facegrps'
*/Users/rmfuhrer/Library/Application Support/SketchUp 2023/SketchUp/Plugins/facesvg/su_util.rb:176:in `block in capture_faceprofiles'
*/Users/rmfuhrer/Library/Application Support/SketchUp 2023/SketchUp/Plugins/facesvg/su_util.rb:128:in `each'
*/Users/rmfuhrer/Library/Application Support/SketchUp 2023/SketchUp/Plugins/facesvg/su_util.rb:128:in `capture_faceprofiles'
*/Users/rmfuhrer/Library/Application Support/SketchUp 2023/SketchUp/Plugins/facesvg/layout.rb:121:in `layout_facegrps'
*/Users/rmfuhrer/Library/Application Support/SketchUp 2023/SketchUp/Plugins/facesvg/layout.rb:88:in `process_selection'
*/Users/rmfuhrer/Library/Application Support/SketchUp 2023/SketchUp/Plugins/facesvg/main.rb:71:in `block in facesvg_2d_layout'
*/Users/rmfuhrer/Library/Application Support/SketchUp 2023/SketchUp/Plugins/facesvg/su_util.rb:46:in `su_operation'
*/Users/rmfuhrer/Library/Application Support/SketchUp 2023/SketchUp/Plugins/facesvg/main.rb:71:in `facesvg_2d_layout'
*/Users/rmfuhrer/Library/Application Support/SketchUp 2023/SketchUp/Plugins/facesvg/main.rb:131:in `block (2 levels) in <module:FaceSVG>'

I'm no Ruby guy, but some of that looks like the plugin is referencing non-existent SketchUp symbols.

Does FaceSVG need an update to work with SketchUp Pro 2023?

**Comment by rmfuhrer** [comment]
In case it helps reproduce the problem, I attached the Sketchup file (gzipped).

[Benchtop Dog Holes.skp.gz](https://github.com/marvingreenberg/FaceSVG/files/13031622/Benchtop.Dog.Holes.skp.gz)

**Comment by marvingreenberg** [comment]
I really haven't had time to look at that code for a while.  I don't think
that the issue is sketch up 2023... I think I've tried it with that.   But
depending on how you create geometries like intersecting lines to get
alignments, and then deleting the lines, I've seen that behavior (use
guides instead).  But it doesn't seem like you would have done that.

But since it sounds like you're just trying to generate holes for an mft
style table, you could check out this post,
https://community.shapertools.com/t/generate-on-the-fly-through-holes-or-pocket-hole-svg-s-for-mft-tops/655

Which has this link
http://dve2000.pythonanywhere.com/holes

That will generate SVG for whatever layout you want, I think.

I wish the code i wrote (and sketchup) worked more consistently.  I still
use it when I'm doing things, but I also haven't been doing much in some
time.

Marvin

**Comment by rmfuhrer** [comment]
Sorry, I took another look at the error messages, and see now that the "Reference to deleted Entity" refers to a model entity, as you say, not a code symbol.

In creating this model, I didn't create any lines (guide or otherwise) to align things. That said, I did go through a few iterations of creating the circle grid. One or more times I probably deleted objects rather than undoing their creation.

You seem to be implying that deleting an object leaves detritus in the model? (I recall that there's a SketchUp command to "garbage collect" unreferenced components from the design, but presumably that's unrelated.)

Anyway, thanks for the pointers. I may give the SVG "hole generator" a shot for the current work. That said, when Shaper Origin added the helix cut mode, it obviated a lot of the functionality of that generator, so now it just generates a grid of circles. I could easily write code to do that myself. (I'm also a sw dev.)

That said, I think your plugin is still a great piece of functionality to have in good working order. :) So, given that I'm a sw dev myself, I may spend some time looking into the SDK, API and such to see about diagnosing the problem further.

**Comment by dzenitis** [comment]
Coming late to this, and this is weird, but I got much the same error recently, and somehow now it is fixed. For a year or more things were working on my old PCs, desktop and laptop. I had to get a new PC, and when I went to install my SketchUp Pro 2021, I was reminded that it was no longer supported and I couldn't get to the Extension Warehouse. I installed the facesvg-3.0.2.rbz that I had saved, since it is one of the most important extensions for me (thank you Marvin!!), and FaceSVG gave that error with Layout SVG profile. I uninstalled, then _by chance_ installed layers.rbz by Joe Zeh (before Layers were called Tags), restarted, ran some of the Layer Tools to test OK, then reinstalled facesvg and restarted. I probably did some other things as well, I know I was also tweaking Preferences, but when I tried FaceSVG again **it worked**!!
So there is hope. Unfortunately for others I don't know why it started working, and I don't have time to do a bunch of testing. And this is a long time after the original post.
